ls-ui-react-app/src/pages/UploadCircuitPage.tsx

```diff
@@ -230,38 +238,40 @@ const UploadCircuitPage = ( {appState, setAppState} : AppStateProps)  =>
                             <b>Processing...</b>
                         </div>
                     }
-
-                    {/* If a request has been made, and compilation has finished, and compilation is still undefined, indicate error message */}
-                    { (appState.request === true && appState.compilationIsLoading === false && appState.compilationResult === undefined) && 
+                    
+                    {/* No response errors */}
+                    { appState.apiError && 
```

Comment:
this will be something like
```ts
appState.apiResponse instanceof ApiHttpError &&
```

And then you can access appState.apiHttpError.msg for the error message

What was previously `appState.apiResponse.status` becomes appState.apiHttpError.code